src/Nancy.Swagger.Annotations/SwaggerObjects/AnnotatedOperation.cs

```diff
@@ -44,21 +44,33 @@ public AnnotatedOperation(string name, MethodInfo handler, ISwaggerModelCatalog
                 .Select(CreateSwaggerResponseObject)
                 .ToDictionary(x => x.GetStatusCode().ToString(), y => (global::Swagger.ObjectModel.Response) y);
 
+            var infos = handler.GetParameters().Where(x => x.GetCustomAttributes<RouteParamAttribute>().Any()).ToList();
 
-            Parameters = handler.GetParameters().Where(x => x.GetCustomAttributes<RouteParamAttribute>().Any())
-                .Select(CreateSwaggerParameterData)
-                .ToList();
+            Parameters = CreateSwaggerParameters(infos);
 
         }
 
-        private Parameter CreateSwaggerParameterData(ParameterInfo pi)
+        private IEnumerable<Parameter> CreateSwaggerParameters(IEnumerable<ParameterInfo> infos)
         {
-            var param = (Parameter)new AnnotatedParameter(pi);
-            if (param.In == ParameterIn.Body)
+            var result = new List<Parameter>();
+            foreach (var info in infos)
             {
-                param = new AnnotatedBodyParameter(pi, _modelCatalog);
+                var paramAttrs = info.GetCustomAttributes<RouteParamAttribute>();
+                //Body param trumps all other attributes.
+                var bodyParamAttr = paramAttrs.FirstOrDefault(x => x.ParamIn == ParameterIn.Body);
+                if (bodyParamAttr != null)
+                {
+                    result.Add(new AnnotatedBodyParameter(info, _modelCatalog));
+                    continue;
```

Comment:
Is this `continue` necessary?

Reply:
The attributes are on a parameter, makes no sense to specify multiple paramin types, it's either body or path or query. At least that's how I see it.

Reply:
D'oh I understand now, overlooked the part where `info` is passed in to `AnnotatedBodyParameter`